simulation/tests/test_agents.py

```diff
@@ -71,3 +73,22 @@ def test_create_household(self) -> None:
         assert household.is_heat_pump_suitable_archetype
         assert household.heating_fuel == HeatingFuel.ELECTRICITY
         assert household.is_heat_pump_aware
+
+    def test_household_renovation_budget_increases_with_property_value(self) -> None:
+        low_property_value_household = household_factory(property_value=100_000)
+        medium_property_value_household = household_factory(property_value=300_000)
+        high_property_value_household = household_factory(property_value=500_000)
+
+        assert (
+            low_property_value_household.renovation_budget
+            < medium_property_value_household.renovation_budget
+            < high_property_value_household.renovation_budget
+        )
+
+    def test_renovation_budget_greater_than_or_equal_to_zero_and_less_than_total_property_value(
+        self,
+    ) -> None:
+        household = household_factory(property_value=random.randint(0, 5_000_000))
```

Comment:
Will a house ever have 0 value? We could change this test to select a `randint(1, 5_000_000)` and then just `assert 0 < household.renovation_budget < household.property_value`.

Also just double checking the behaviour here: we are assuming that _all_ houses will have a renovation budget (equal to, currently, 10% of their property value) regardless of any other attributes, for now.

Reply:
I think DQ of the source data should primarily be covered within data warehouse - e.g. non-zero property values - rather than kind of mixed into this test.

Not quite - all houses have a renovation budget, which is drawn from a distribution of GB renovation budgets at the same percentile as their relative property value. So this deliberately doesn't need to correspond to 10% of their property value (in practice ranges between 0-25% for most households). 10% of this overall budget is allocated by a house to "heating systems and energy efficiency measures" - the part of the budget relevant to the future heating system decision logic.